<launch>

<node pkg="tf" type="static_transform_publisher" name="origin_marker_to_tf" args="0 0 0 0 0 -1.57  aruco_marker_1 map 50"/>
<node pkg="tf" type="static_transform_publisher" name="bake_link_marker_to_tf" args="0 0 0 0 0 -1.57  aruco_marker_2 base_link 50"/>


	<!--remote control -->	
	<node pkg="rc_bringup" name="rc_node" type="rc_control.py" respawn="true" output="screen">
	<param name="cmd_vel" value="cmd_vel"/>
	<param name="pwm_topic" value="pwm"/>
	<param name="drive_topic" value="ackermann_cmd"/>
	<param name="max_vel" value="2.4"/>
	<param name="min_vel" value="1.0"/>
	<param name="wheelbase" value="0.28"/>
	<param name="max_steering_angle" value="25.0"/>
	<param name="servo_pin" value="4"/>
	<param name="middle_servo" value="1500"/>
	<param name="servo_offset" value="47"/>    
	<param name="motor_pin" value="17"/>
	<param name="middle_motor" value="1550"/>
	<param name="revers_servo" value="False"/>
	</node>


	<!--pose controller -->
	<node pkg="rc_bringup" name="rc_pose_controller" type="pose_controller.py" respawn="true" output="screen">
		<param name="vel_topic" value="velocity"/>
		<param name="cmd_vel" value="cmd_vel"/>
		<param name="goal_topic" value="move_base_simple/goal"/>
		<param name="base_link" value="map"/>
		<param name="child_link" value="base_link"/>

		<param name="max_vel" value="1.0"/>
		<param name="min_vel" value="-1.5"/>
		<param name="max_angle" value="25.0"/>
		<param name="goal_tolerance" value="0.4"/>

		<param name="kP_pose" value="1.0"/>
		<param name="kI_pose" value="0.0"/>
		<param name="kD_pose" value="0.2"/>

		<param name="kP_course" value="0.5"/>
		<param name="kI_course" value="0.0"/>
		<param name="kD_course" value="0.2"/>
	</node>





</launch>
